snap/info_snap_yaml.go

```diff
@@ -292,6 +292,12 @@ func setAppsFromSnapYaml(y snapYaml, snap *Info) {
 
 func setHooksFromSnapYaml(y snapYaml, snap *Info) {
 	for hookName, yHook := range y.Hooks {
+		// Verify that the requested hook name is actually supported. If not,
+		// ignore it.
+		if !IsHookSupported(hookName) {
+			continue
```

Comment:
There's nothing necessarily wrong with a snap containing a hook that the local system does not support.